### PR TITLE
[Fix #2070] Harnessing TestEventPublisher

### LINKED ISSUE #2070: EmbeddedJobServiceTest potentially flaky
### Description

At least one PR was failing in Jenkins two weeks ago with this error 
https://github.com/apache/incubator-kie-kogito-apps/pull/2068#issuecomment-2141542284
After analysing the test code, some improvements can be done to avoid it. 

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by fjtirado**
Fix https://github.com/apache/incubator-kie-kogito-apps/issues/2070

### CODE REVIEW

File: jobs-service/kogito-addons-jobs-service/kogito-addons-quarkus-jobs/src/test/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1 @@
+quarkus.devservices.enabled=false
```
Comment (fjtirado) [inline jobs-service/kogito-addons-jobs-service/kogito-addons-quarkus-jobs/src/test/resources/application.properties:1]: This test does not really need quarkus dev service to be enable (or at least I believe so, lets see CI)

**Review by pefernan — COMMENTED** [review]
+1 waiting for the green

**Comment by kie-ci3** [comment]
**PR job** `#2` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-apps -u https://github.com/apache/incubator-kie-kogito-apps/pull/2071 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-apps-pr/job/PR-2071/2/display/redirect

**Test results:**
- PASSED: 1663
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-apps-pr/job/PR-2071/2/testReport/org.kie.kogito.index/ProcessDataIndexInfinispanIT/testProcessInstanceEvents/">org.kie.kogito.index.ProcessDataIndexInfinispanIT.testProcessInstanceEvents</a></summary>
1 expectation failed.<br/>JSON path errors doesn't match.<br/>Expected: null<br/>  Actual: <[{message=Exception while fetching data (/UserTaskInstances[0]/schema) : Runtime service URL not defined, please review the kogito.service.url system property to point the public URL for this runtime., locations=[{line=1, column=85}], path=[UserTaskInstances, 0, schema], extensions={classification=DataFetchingException}}]><br/>
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-apps-pr/job/PR-2071/2/testReport/org.kie.kogito.index/ProcessDataIndexMongoDBIT/testProcessInstanceEvents/">org.kie.kogito.index.ProcessDataIndexMongoDBIT.testProcessInstanceEvents</a></summary>
Cannot invoke method size() on null object
</details>

**Comment by fjtirado** [comment]
@pefernan There are unrelated failures. I propose to merge this one and create an issue for the other two. Wdyt?

**Comment by pefernan** [comment]
@fjtirado we can go for it!

**Review by elguardian — APPROVED** [review]
thx @fjtirado for taking care of this.

**Pull request merged**